maro/event_buffer/event_pool.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from .atom_event import AtomEvent
+from .cascade_event import CascadeEvent
+from .event_state import EventState
+from .typings import Event, EventList, List, Union
+
+
+class EventPool:
+    """Event pool used to generate and pool event object.
+
+    The pooling function is disabled by default, then it is used as an Event generator with a buffer.
+
+    When enable pooling, it will recycle events.
+    """
+
+    def __init__(self, enable: bool = False, capacity: int = 1000):
+        self._capacity = capacity
+        self._enabled: bool = enable
+        self._atom_pool: List[AtomEvent] = []
+        self._cascade_pool: List[CascadeEvent] = []
+        self._recycle_buffer: EventList = []
+
+        self._event_id: int = 0
+
+        if enable:
+            for _ in range(capacity):
+                self._atom_pool.append(AtomEvent(None, None, None, None))
```

Comment:
why initialize #capacity None Events when the pool is empty? I think it may waste some space but can this improve the performance or anything?

Reply:
changed, create instance if needed or pop